Assets/Scripts/GameController.cs

```diff
@@ -128,28 +132,99 @@ public class GameController : MonoBehaviour {
 		}
 	}
 
+	// Token actions when unit is placed
+	public void PlaceUnit(Token token) {
+		token.CurrentUnit = SC.CreateUnit(UnitBeingPlaced.matchUnit,token.X,token.Y, true);
+	}
+
 	// Runs when a unit is selected. Take action if possible, otherwise show unit info
 	public void SelectUnit(Token token) {
 		Unit unit = token.CurrentUnit;
+		ShowUnitInfo(unit);
 		SelectedToken = token;
 		if(unit.MyTeam && !unit.TakenAction && GameData.CurrentMatch.UserTurn) {
 			SetValidActions(token);
 		}else{
-			ShowUnitInfo(unit);
+			
 		}
 	}
 
 	// Placeholder for what will contain code to show unit's info on UI
 	public void ShowUnitInfo(Unit unit) {
-		UnityEngine.Debug.Log(((unit.MyTeam)? "Your " : "Opponent's ") + unit.Info.Name);
+		UnitInfo.SetUnitInfo(unit.Info);
 	}
 
 	// Runs when a unit is unselected (i.e. user clicks other unit, or unit takes turn)
 	public void UnselectUnit() {
+		UnitInfo.RemoveUnitInfo();
 		SelectedToken = null;
+		IntendedMove = null;
+		IntendedAttack = null;
 		ClearValidActions();
 	}
 
+	// Repaints the available actions after a unit's intended move is set
+	public void PaintIntendedMoveActions() {
+		foreach(KeyValuePair<Twople<int, int>, int> action in Actions) {
+			Token currToken = Tokens[action.Key.Item1][action.Key.Item2];
+			if(currToken == IntendedMove || CanAttackFromToken(currToken)) {
+				continue;
+			}
+			currToken.SetActionProperties("clear");
+		}
+	}
+
+	public bool CanAttackFromToken(Token currToken) {
+		if(currToken.CanAttack && currToken.HasEnemy) {
+			int _x = (IntendedMove != null)? Mathf.Abs(IntendedMove.X - currToken.X) : Mathf.Abs(SelectedToken.X - currToken.X);
+			int _y = (IntendedMove != null)? Mathf.Abs(IntendedMove.Y - currToken.Y) : Mathf.Abs(SelectedToken.Y - currToken.Y);
+			int atkRange = GameData.GetUnit(SelectedToken.CurrentUnit.Info.Name).GetStat("Attack Range").Value;
+			return atkRange >= _x + _y;
+		}
+		return false;
+	}
+
+	// Move unit to new token and paint new actions
+	public void SetIntendedMove(Token token) {
+		IntendedMove = token;
+		SelectedToken.CurrentUnit.transform.position = token.gameObject.transform.position;
+		PaintIntendedMoveActions();
+	}
+	// Confirm move unit to new token and unselect after
+	public void ConfirmMove() {
+		if(Server.TakeMoveAction(SelectedToken.CurrentUnit,"Wait", IntendedMove.X, IntendedMove.Y)) {
+			MoveUnit();
+		}
+	}
+	// All the actions when moving a unit
+	public void MoveUnit() {
+		IntendedMove.CurrentUnit = SelectedToken.CurrentUnit;
+		SelectedToken.CurrentUnit = null;
+		IntendedMove.CurrentUnit.ConfirmMove();
+	}
+	public void SetIntendedAttack(Token token) {
+		IntendedAttack = token;
+		//Insert function to show attack info
+	}
+	// Confirms attack target and moves to intended token if applicable
+	public void ConfirmAttack(Unit targetUnit) {
```

Comment:
what is that? lol jk I just didn't add healing - will make a 2nd pr to account for that